lib/util/exists.js

```diff
@@ -50,7 +50,9 @@ module.exports = function exists(filePath) {
                 fs.statSync(relativePath).isFile() &&
                 existsCaseSensitive(relativePath)
         } catch (error) {
-            if (error.code !== "ENOENT") {
+            if (
+                /** @type {NodeJS.ErrnoException} */ (error).code !== "ENOENT"
+            ) {
```

Comment:
Instead of type casting I'm using a small helper in my own code: https://github.com/voxpelli/list-installed/blob/137ea5e81971221f80d847477ac88dc634135900/lib/utils.js#L21-L27

```js
/**
 * @param {unknown} value
 * @returns {value is NodeJS.ErrnoException}
 */
export function looksLikeAnErrnoException (value) {
  return value instanceof Error && 'code' in value && 'path' in value;
}
```

```suggestion
            if (!looksLikeAnErrnoException(error) || error.code !== "ENOENT") {
```